### PR TITLE
Feat: add web terminal

### PR DESCRIPTION
**Pull request by harlee-x**
### Description of your changes

Fixes #add web terminal for kdp

### CODE REVIEW

**Review by jason-ltc — CHANGES_REQUESTED** [review]

File: infra/kdp-core/parameter.cue [hunk 1]
```diff
@@ -46,7 +46,13 @@ parameter: {
 			}
 		}
 	}
-
+	// +usage=Specify kdp web name
+	webName: *"kdp-ux" | string
```
[the remark this answers is not in the document]
Comment (harlee-x) [reply infra/kdp-core/parameter.cue]: Using a fixed parameter, this parameter cancels exposure

File: infra/kdp-core/parameter.cue [hunk 2]
```diff
@@ -46,7 +46,13 @@ parameter: {
...
+	// +usage=Specify kdp web port
+	webPort: *3300 | int
+	// +usage=Specify web terminal Three-level domain name
+	terminalHost: *"cloudtty" | string
```
[the remark this answers is not in the document]
Comment (harlee-x) [reply infra/kdp-core/parameter.cue]: Using a fixed parameter, this parameter cancels exposure

File: infra/kdp-core/parameter.cue [hunk 3]
```diff
@@ -46,7 +46,13 @@ parameter: {
...
+	PROXY_KEY: *"KONG_KONG_PROXY_SERVICE_HOST" | string
```
[the remark this answers is not in the document]
Comment (harlee-x) [reply infra/kdp-core/parameter.cue]: delete it

File: infra/kdp-core/parameter.cue
Comment (jason-ltc) [inline infra/kdp-core/parameter.cue:1]: This is unrelated.

File: infra/kdp-core/resources/kdp-cloudtty.cue [hunk 4]
```diff
@@ -0,0 +1,447 @@
...
+
+_kdpCloudTty: {
+	name: parameter.namePrefix + _CloudTtyName
+	type: "helm"
+	properties: {
+		url:             "\(parameter.helmURL)"
+		chart:           _CloudTtyName
+		releaseName:     parameter.namePrefix + _CloudTtyName
+		repoType:        "oci"
+		version:         "0.5.7"
```
Comment (jason-ltc) [inline infra/kdp-core/resources/kdp-cloudtty.cue:13]: L13 and L77 use the same value, we should define a variable for them.

File: infra/kdp-core/resources/kdp-cloudtty.cue [hunk 5]
```diff
@@ -0,0 +1,447 @@
...
+					{
+						apiGroups: ["cloudshell.cloudtty.io"]
+						resources: [
+							"cloudshells"
+						]
+						verbs: [
+							"get",
+							"list",
+						]
+					},
```
Comment (jason-ltc) [inline infra/kdp-core/resources/kdp-cloudtty.cue:319]: Shall we add those resources controlled by big-data components such as flink/kafka/spark operators too?

**Review by jason-ltc — CHANGES_REQUESTED** [review]

File: infra/kdp-core/resources/kdp-cloudtty.cue (hunk 4 above)
Comment (harlee-x) [reply infra/kdp-core/resources/kdp-cloudtty.cue:13]: chart values and image values are not the same, so the same variable cannot be used here

**Review by jason-ltc — APPROVED** [review]
